### PR TITLE
nixos/doc/release-process: remove

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
The related documentation on the release process has moved
to https://github.com/NixOS/release-wiki

Doesn't make sense to export the release process to users.

cc @worldofpeace 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
bump, @worldofpeace

**Comment by jtojnar** [comment]
Would it make sense to al least link the wiki in the manual?

**Comment by jonringer** [comment]
It would, let me add that

**Comment by jonringer** [comment]
done

File: nixos/doc/manual/development/releases.xml [hunk 1]
```diff
@@ -8,359 +8,9 @@
   <title>Release process</title>
 
   <para>
-   Going through an example of releasing NixOS 19.09:
+   This section has been moved to <link xlink:href="https://github.com/NixOS/release-wiki">the release wiki</link>.
```
Comment (worldofpeace) [inline nixos/doc/manual/development/releases.xml:11]: It doesn't make sense to link it to the wiki like this. Instead there needs to be a section explaining the how and why of the release process, and then say "If you're interested in how a NixOS release actually happens please see https://nixos.github.io/release-wiki/` and that's the rendered link. Speaking of which the github action is failing there.

Comment (jonringer) [reply nixos/doc/manual/development/releases.xml:11]: I wouldn't really expect for the release process to be there in the first place. I think this mostly benefits people who are used to it being there. 

New users will probably find it confusing that it's mentioned at all.

**Comment by siraben** [comment]
Any updates on this? Looks like it's a good idea to remove unnecessary info about release process in the manual.

**Comment by jonringer** [comment]
let me deal with conflicts

**Comment by jonringer** [comment]
Nevermind. This is a duplicate of https://github.com/NixOS/nixpkgs/pull/124418 now. closing

**Pull request closed without merging**